ide-common/src/main/java/org/digma/intellij/plugin/document/DocumentInfoContainer.java

```diff
@@ -69,16 +70,26 @@ public void refresh() {
         loadAllInsightsForCurrentEnvironment();
     }
 
+    public List<CodeObjectInsight> getMethodInsights(@NotNull MethodInfo methodInfo) {
+        Log.log(LOGGER::debug, "Requesting insights for MethodInfo {}", methodInfo.getId());
+
+        return insights.stream().filter(codeObjectInsight -> {
+            String codeObjectId = codeObjectInsight.getCodeObjectId();
+            return methodInfo.getId().equals(codeObjectId) || methodInfo.getRelatedCodeObjectIds().contains(codeObjectId)
+                    || methodInfo.getRelatedCodeObjectIdsWithType().contains(codeObjectId);
```

Comment:
if you check getRelatedCodeObjectIdsWithType then maybe you need to check also methodInfo.getIdWithType ?

Reply:
thanks! added this also